### DISCUSSION TITLE
Using BDFR in python files

### DISCUSSION THREAD

**Discussion by CZAlmon** [Serene-Arc/bulk-downloader-for-reddit#587] [Q&A] [answered]
So I am following along with the README, and have been able to get the command line working. From the README, all the commands are geared towards just working from the command line, and even the tests use bash scripts to run them. Is it possible to run BDFR from within a python file? I know this is a dumb question, I just have been out of programming with python for a while and I have always learned better by seeing examples and working from there. Is it possible to import BDFR and run the download commands from within a python file?

**Comment by Serene-Arc** [comment, marked as answer]
No, that's not how it works, and I'm not quite sure what you're trying to do. The commands aren't run in a Python file, they're run in the CLI. That's the only way they're supposed to be used, trying to use them in a Python file doesn't make sense. It's like trying to get a game to output to Microsoft Word; they're not supposed to be compatible.

If you mean using the BDFR's classes and code as an interface, then the commands are irrelevant, don't worry about them. However we don't have an API reference so you'll have to go through the code and find the classes and functionality that you need.

**Reply by CZAlmon** [reply]
That answers my questions. Would be cool to be able to build a program around the commands so you could input different subreddits or different commands (download/archive/clone) and to be able to automate it to run at a timed interval. 

That was my idea anyways. I guess you can do the same thing with the bash scripts, my only thought was trying it in python, since BDFR is already made in python.

Anyways thanks for the answer!

**Reply by starmadegeek** [reply]
Using bash scrits can be a pain. I also wanted the reddit posts download functinality in a personal app that I'm building for myself in python. If you tried and succeeded in using BDFR in python scripts, please share the usage with me. Thanks!

**Reply by Serene-Arc** [reply]
This is possible. You can import the classes and use the functions like any library, I just don't have any written documentation for it. Also, I can't promise that it won't change.
